### PR TITLE
implemented few tests for user route + added few todo tests

### PR DESCRIPTION
**Pull request by iritmaximus**

### CODE REVIEW

File: backend/src/api-tests/user/create.test.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { beforeAll, afterAll, describe, it } from '@jest/globals'
+import { resetDatabase, resetDatabaseTimeout } from '../utils'
+import { pool } from '../../utils/db'
+
+describe('Creating an user', () => {
```
Comment (realtalin) [inline backend/src/api-tests/user/create.test.ts:5]:
```suggestion
describe('Creating a user', () => {
```

File: backend/src/api-tests/user/create.test.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  beforeAll(async () => {
+    await resetDatabase()
+  }, resetDatabaseTimeout)
+  afterAll(async () => {
+    await pool.end()
+  })
+
+  it.todo('is successful with correct data')
+  it.todo('is not successfull with missing data')
```
Comment (realtalin) [inline backend/src/api-tests/user/create.test.ts:14]:
```suggestion
  it.todo('is not successful with missing data')
```

File: backend/src/api-tests/user/create.test.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  it.todo('is not successfull with incorrect password')
```
Comment (realtalin) [inline backend/src/api-tests/user/create.test.ts:15]:
```suggestion
  it.todo('is not successful with incorrect password')
```

File: backend/src/api-tests/user/login.test.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  it('is successful with test admin user', async () => {
+    const result = await send('user/login', 'POST', testSu)
+
+    expect(result.status).toEqual(200)
+    expect(result.body.username).toEqual('testSu')
+    expect(result.body.role).toEqual(1) // frontend/src/types: Role enum
+    expect(result.body.isFirstLogin).toEqual(undefined)
+  })
+
+  it('is not successfull with missing data', async () => {
```
Comment (realtalin) [inline backend/src/api-tests/user/login.test.ts:23]:
```suggestion
  it('is not successful with missing data', async () => {
```

File: backend/src/api-tests/user/login.test.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    expect(resultNoUsername.status).toEqual(403)
+    expect(resultNoPassword.status).toEqual(403)
+
+    const resultEmptyUsername = await send('user/login', 'POST', { username: '', password: 'test' })
+    const resultEmptyPassword = await send('user/login', 'POST', { username: 'testSu', password: '' })
+    expect(resultEmptyUsername.status).toEqual(403)
+    expect(resultEmptyPassword.status).toEqual(403)
+  })
+
+  it('is not successfull with incorrect password', async () => {
```
Comment (realtalin) [inline backend/src/api-tests/user/login.test.ts:35]:
```suggestion
  it('is not successful with incorrect password', async () => {
```

**Review by realtalin — CHANGES_REQUESTED** [review]
sorry for pointing out grammar issues but it's going to bug me if i don't 😅

otherwise looks good!

**Comment by iritmaximus** [comment]
Done.

**Review by realtalin — APPROVED** [review]

**Pull request merged**